apps/web/server/routers/viewer.tsx

```diff
@@ -936,6 +937,104 @@ const loggedInViewerRouter = createProtectedRouter()
 
       return locationOptions;
     },
+  })
+  .mutation("deleteCredential", {
```

Comment:
Over all very nice progress! One more thing tho: Since we're doing multiple queries we should run them in [a transaction](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#advanced-example-update-a-nested-json-key-value), doing so will prevent us to be left with incomplete data if any of the queries fail